calicoctl/calico_ctl/node.py

```diff
@@ -847,8 +848,13 @@ def handle_sigterm(sig, frame):
     except KeyboardInterrupt:
         # Mainline. Someone pressed Ctrl-C.
         print "Stopping Calico node..."
+        clean_exit = True
     finally:
         # Could either be this process is being killed, or output generator
         # raises an exception.
         if stop_container_on_exit:
             docker_client.stop(container)
+            # If the container is stopped due to unknown circumstances, exit
+            # with failed status code.
+            if not clean_exit:
+                sys.exit(1)
```

Comment:
Actually, I'm thinking `attach` vs `wait` a bit more and doing some minor tests to check if it would work